command/base.go

```diff
@@ -589,7 +589,7 @@ func (f *FlagSets) Parse(args []string) error {
 	err := f.mainSet.Parse(args)
 
 	warnings := generateFlagWarnings(f.Args())
-	if warnings != "" {
+	if warnings != "" && Format(f.ui) != "json" {
```

Comment:
Should we also exclude yaml? Or maybe it's simpler to have an allow list, probably table and pretty? Full list of supported formatters here: https://github.com/hashicorp/vault/blob/7d08dcc21a1d4cd6f7974e3e60408b1e4d43fec9/command/format.go#L67-L73

Reply:
+1

Also, would it be possible to capture these scenarios in some tests? Unit or integration?

Reply:
Do you have a link or example of integration tests that validate the format of some outputs?

Reply:
Looks like pretty is for custom formatters. Currently it is only implemented for the [raft autopilot state API](https://github.com/hashicorp/vault/blob/195e025cb2646209ccd49caef395690cbcdb7bd3/command/format.go#L151)

Reply:
> Do you have a link or example of integration tests that validate the format of some outputs?

I realize many of these cases are more like unplanned "features" that just happened as vault grew. And are probably very unlikely edge cases. So it may not be worthwhile.